src/jit/flowgraph.cpp

```diff
@@ -13476,6 +13479,9 @@ bool Compiler::fgOptimizeEmptyBlock(BasicBlock* block)
                         if (block->IsLIR())
                         {
                             LIR::AsRange(block).InsertAtEnd(nop);
+#ifndef LEGACY_BACKEND
+                            m_pLowering->LowerRange(nop, nop);
```

Comment:
As mentioned offline, reusing the same lowering instance for all of these calls to `LowerRange` is not safe without first resetting its state. I would recommend making `LowerRange` a static method that creates a new `Lowering` instance on the stack for each call (this is the approach taken by `DecomposeRange`).